Dockerfile

```diff
@@ -39,14 +42,26 @@ RUN apt-get update -qq && \
 
 # Install the latest nightly Clang/lld packages from apt.llvm.org
 RUN curl https://apt.llvm.org/llvm-snapshot.gpg.key | apt-key add - && \
-    echo "deb http://apt.llvm.org/unstable/ llvm-toolchain$(test ${LLVM_VERSION} -ne 9 && echo "-${LLVM_VERSION}") main" | tee -a /etc/apt/sources.list && \
+    echo "deb http://apt.llvm.org/unstable/ llvm-toolchain$([[ ${LLVM_VERSION} -ne 9 ]] && echo "-${LLVM_VERSION}") main" | tee -a /etc/apt/sources.list && \
     apt-get update -qq && \
     apt-get install --no-install-recommends -y \
         clang-${LLVM_VERSION} \
         lld-${LLVM_VERSION} \
         llvm-${LLVM_VERSION} && \
     chmod -f +x /usr/lib/llvm-${LLVM_VERSION}/bin/*
 
+# Check and see Clang has not been rebuilt in more than five days if we are on the master branch and fail the build if so
+RUN if [[ ${LLVM_VERSION} -eq 9 ]]; then \
+    DATE=$(clang-${LLVM_VERSION} --version | head -n1 | cut -d '~' -f 3 | cut -d . -f 1) && \
+    DATE="${DATE:0:4}-${DATE:4:2}-${DATE:6:2} ${DATE:8:2}:${DATE:10:2}:${DATE:12:2}" && \
+    DAYS_SINCE=$(($(($(date -u +%s) - $(date -u -d "${DATE:?}" +%s))) / 86400)) && \
+    if [[ ${DAYS_SINCE} -ge 5 ]]; then \
+        echo; \
+        echo "clang-${LLVM_VERSION} has not been updated in ${DAYS_SINCE} days!"; \
+        echo; \
+        exit 1; \
+    fi; fi
```

Comment:
can this logic be put in a shell script, with proper functions (`days_since` could be a function) and comments, since there's quite a few complicated expressions here that would be difficult for others to maintain?

Reply:
Sure I will check.